Classes/Emogrifier.php

```diff
@@ -464,6 +466,85 @@ private function mapAllInlineStylesToHtmlAttributes(\DOMXPath $xPath)
     }
 
     /**
+     * Searches for all nodes with a style attribute and removes the "!important" annotations out of
+     * the inline style declarations, eventually by rearranging declarations.
+     *
+     * @uses \Pelago\Emogrifier::removeImportantAnnotationFromNodeInlineStyle
+     *
+     * @param \DOMXPath $xPath
+     *
+     * @return void
+     */
+    private function removeImportantAnnotationFromAllInlineStyles(\DOMXPath $xPath)
+    {
+        $nodesWithStyleAttribute = $this->getAllNodesWithStyleAttribute($xPath);
```

Comment:
As this method is the only method calling getAllNodesWithStyleAttribute, we could always throw the exception there and then catch it here and either do an early return or rethrow the exception in debug mode.

Reply:
OK.
As we are talking about the method `getAllNodesWithStyleAttribute()`, I am planning to use it in two other places, where `$xPath->query('//*[@style]')` is now directly used. What do you think? Should I do these modifications within this PR or submit another one after this is merged?